src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -148,6 +157,8 @@ public Http2Connection(HttpConnectionPool pool, Connection connection)
             }
         }
 
+        ~Http2Connection() => Dispose();
```

Comment:
We have complete ownership of it at all times, but it's not guaranteed that Dispose will ever be called if the HttpClient/Handler is never disposed of as there is no finalizer higher up in the chain.

We could instead introduce a finalizer higher in the chain (for example on ConnectionPoolManager) to avoid having one per-Http2Connection, but such a finalizer isn't needed for any other use-case / for HTTP/1 at the moment afaik.